### PR TITLE
refactor(Build): Allow UMD packed module to run in the browser

### LINKED ISSUE #178: Schema v0.38.0+ are not supported in browser environments
As can be seen in [build log for v0.38.0](https://travis-ci.org/github/stencila/schema/jobs/653733345#L310), the changes in that version introduce dependency on `fs` and `path` Node modules.

> ```
> Creating a browser bundle that depends on Node.js built-in module ('path').
> You might need to include https://www.npmjs.com/package/rollup-plugin-node-builtins
> ```

There is a need to run several of the schema node factory functions (`codeChunk({…})`, etc.), on the front end. With the above changes it is not possible to build those packages without adding bloat with the polyfills, which I'm honestly not sure would function out of the box.

A relatively simple first step could be to omit `jsonld`, `jsonSchemas`, and `microdata` packages from browser builds. A bit more involved refactor could include removing the dynamic file reads, but not sure it'll be possible for all functions without trying (`microdata` file should be relative simple though) .

https://github.com/stencila/schema/blob/a7ae82a9fd6cd35c849ff6f8a2434487dba1bc77/ts/util/index.ts#L1-L6

### PR DESCRIPTION
**Pull request by alex-ketch**
Closes #178 

This PR swaps out the build tooling to [TSDX](https://github.com/jaredpalmer/tsdx) which is a tool I came across after originally setting up Microbundle. I made a PR to Redux Toolkit which uses it, and like the ability to provide overrides for Rollup if needed.

This uses similar approach as Redux Toolkit of custom comment markers to conditionally [strip code](https://github.com/se-panfilov/rollup-plugin-strip-code) based on build target.
Alternatively we can maintain multiple entry point files to pass into TSDX.

I've tested this build locally in Encoda and Designa to confirm that it still works, but as always extra eyes/hands on testing is appreciated.

There is a [change required in Encoda here](https://github.com/stencila/encoda/blob/2acae92f35cf7dd1a84547298fdf4874e9133129/src/codecs/html/html.test.ts#L1) due to conflicting ways we've been using default imports, and TypeScript's attempts to fix it.

```diff
- import stencila, {
+ import * as stencila from '@stencila/schema'
import {
```

Let me know if the above changes sound reasonable to you, and I can make corresponding fix in Encoda, or try the multi-entry build process and stick with Microbundle if you'd like @nokome.

---

Update: There are issues with type resolution in Encoda that are not captured by running tests. Looking into it

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
   ],
   "source": "ts/index.ts",
```
Comment (nokome) [inline package.json]: Can / should `source` be removed?

File: ts/util/version.ts [hunk 2]
```diff
@@ -1,44 +1,19 @@
-import fs from 'fs'
-import path from 'path'
-
-/**
- * Lazily read, cached, package version
- */
-let VERSION: string
+import { version as pkgVersion } from '../../package.json'
```
Comment (nokome) [inline ts/util/version.ts]: Yep, this is obvious, more elegant way to do this. But I have been caught in the past by a gotcha that this creates related to having a nested `package.json` in dist. IIRC, it only rears its head in dependants of this package. I'll check and report back.

**Comment by alex-ketch** [comment]
This issue is turning out to be more of a headache than anticipated and not an immediate blocker for my current work. Will return to it tomorrow/Monday.

In short returning to Microbundle with multiple entries fixes Encoda errors, but introduces new ones with Designa's browser module.

File: ts/util/version.ts (hunk 2 above)
Comment (nokome) [reply ts/util/version.ts]: Checked, and this won't be a problem because no `package.json` ends up in `dist` in any case

File: package.json (hunk 1 above)
Comment (alex-ketch) [reply package.json:8]: `source` is used as the [entry point for Microbundle](https://github.com/developit/microbundle#set-up-your-packagejson)

File: package.json [hunk 3]
```diff
@@ -8,20 +8,24 @@
...
   "main": "dist/index.js",
   "module": "dist/index.mjs",
-  "unpkg": "dist/index.umd.js",
+  "browser": "dist/browser/schema.mjs",
+  "unpkg": "dist/browser/schema.umd.js",
```
Comment (alex-ketch) [inline package.json:12]: Even though the [`browser` field can be a map](https://github.com/defunctzombie/package-browser-field-spec#replace-specific-files---advanced) to handle both modern/module/`mjs` file and the UMD file, it caused issues when used from StencilJS.

The current approach assumes that the `browser` field is used from other project and will go through their own compilation process, and the `unpkg` field will get served through CDN usage and not require any post-processing.

**Pull request merged**